src/store/global/global.thunks.ts

```diff
@@ -708,32 +711,35 @@ function checkPendingDeposits() {
             const successfulTxReceipts = txReceipts.filter(
               (txReceipt) => txReceipt.status === 1 && txReceipt.logs && txReceipt.logs.length > 0
             );
-            const transactionHistoryPromises = successfulTxReceipts.map((txReceipt) => {
-              // Need to parse logs, but only events from the Hermez SC. Ignore errors when trying to parse others
-              const parsedLogs = [];
-              for (const txReceiptLog of txReceipt.logs) {
-                try {
-                  const parsedLog = hermezContractInterface.parseLog(txReceiptLog);
-                  parsedLogs.push(parsedLog);
-                } catch (e) {}
-              }
-              const l1UserTxEvent = parsedLogs.find((event) => event.name === "L1UserTxEvent");
+            const transactionHistoryPromises = successfulTxReceipts.reduce(
+              (accL2Transactions: Promise<L2Transaction>[], txReceipt) => {
+                // Need to parse logs, but only events from the Hermez SC. Ignore errors when trying to parse others
+                const parsedLogs = [];
+                for (const txReceiptLog of txReceipt.logs) {
+                  try {
+                    const parsedLog = hermezContractInterface.parseLog(txReceiptLog);
+                    parsedLogs.push(parsedLog);
+                  } catch (e) {}
+                }
+                const l1UserTxEvent = parsedLogs.find((event) => event.name === "L1UserTxEvent");
 
-              if (!l1UserTxEvent) {
-                return Promise.resolve();
-              }
+                if (!l1UserTxEvent) {
+                  return accL2Transactions;
```

Comment:
Is it ok to skip the transaction when `!l1UserTxEvent`?

Reply:
Yep, in fact that was what `Promise.resolve()` tries to do. If the event hasn't been emitted we need to skip this transaction. I think that we can also remove L741 right?